pkgs/development/python-modules/notmuch2/default.nix

```diff
@@ -20,17 +20,20 @@ buildPythonPackage {
   ];
 
   # since notmuch 0.35, this package expects _notmuch_config.py that is
-  # generated by notmuch's configure script
+  # generated by notmuch's configure script. Substitute our nix-crafted one.
```

Comment:
I'd prefer if we'd either make this hack darwin-only (it's finding the right paths on linux at least), and even better, try to fix it under darwin, rather than writing the file outselves. This will probably get out of sync over time and cause odd bugs.

I did a `nix-build -A notmuch.bindingconfig` and peeked at `_notmuch_config.py`, the paths in there point to `/private/tmp`:

```
NOTMUCH_VERSION_FILE='/private/tmp/nix-build-notmuch-0.37.drv-0/notmuch-0.37/version.txt'
NOTMUCH_INCLUDE_DIR='/private/tmp/nix-build-notmuch-0.37.drv-0/notmuch-0.37/lib'
NOTMUCH_LIB_DIR='/private/tmp/nix-build-notmuch-0.37.drv-0/notmuch-0.37/lib'
```

Maybe some path rewriting logic is missing here, causing the wrong path to be picked up?